pages/content.js

```diff
@@ -20,6 +20,112 @@ export async function getServerSideProps() {
 export default function Content({ currentCollections }) {
   const currentArticle = useStore((state) => state.currentArticle);
   const [popUp, setPopUp] = useState(false);
+  const [selections, setSelections] = useState([]);
+
+  function restoreHighlightRemoveEvent() {
+    const spans = document.querySelectorAll(".highlight");
+    spans.forEach((span) => span.addEventListener("dblclick", removeHighlight));
+  }
+
+  function isSelectionOverlap(newRange, oldRange) {
+    //check if the new selection is in the same paragraphs with the previoous ones
+    if (newRange.commonAncestorContainer !== oldRange.commonAncestorContainer)
+      return false;
+
+    if (
+      newRange.startOffset > oldRange.startOffset &&
+      newRange.startOffset < oldRange.endOffset
+    )
+      return true;
+
+    if (
+      newRange.endOffset > oldRange.startOffset &&
+      newRange.endOffset < oldRange.endOffset
+    )
+      return true;
+
+    if (
+      newRange.startOffset < oldRange.startOffset &&
+      newRange.endOffset > oldRange.endOffset
+    )
+      return true;
+
+    return false;
+  }
+
+  //update document with highlights to database
+  async function saveSelection2DB() {
+    const updatedContent = document.querySelector(
+      ".articleFullContent"
+    ).innerHTML;
+    try {
+      const response = await fetch(`/api/article/${currentArticle.id}`, {
+        method: "PUT",
+        body: JSON.stringify({
+          ...currentArticle,
+          fullContent: updatedContent,
+        }),
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  function removeHighlight(event) {
+    const span = event.target;
+    span.classList.remove(".highlight");
+    const text = span.textContent;
+    span.replaceWith(text);
+    setSelections((prevSels) =>
+      prevSels.filter(
+        (prevSel) => span.getAttribute("id") != prevSel.span.getAttribute("id")
+      )
+    );
+  }
+
+  function highLight(event) {
+    const sel = window.getSelection();
+
+    if (!sel) return; //if no selection
+    if (sel.toString().length == 0) return;
+    if (event.detail === 2) return; // double mouse click
+    const range = window.getSelection().getRangeAt(0);
+    if (
+      range.commonAncestorContainer.tagName !== undefined &&
+      range.commonAncestorContainer.tagName !== "P" &&
+      range.commonAncestorContainer.tagName !== "SPAN"
+    ) {
+      window.alert("Selection over several paragraph is not allowed");
```

Comment:
i know it looked itchy right? :D

Reply:
I changed the condition to range.commonAncestorContainer.tagName ==="DIV" now since i dont know what can come inside a paragraph. As I said earlier, the offset values depend on the nearest tag, and so the container :/. Most of the articles pack div around all paragraphs but maybe some other newspaper decide to pack a section then i m screwed again :(

Reply:
maybe you can find a better solution instead of looking at the commonAncestorContainer? But I guess for now its good.

Reply:
yes maybe i will assign id to every possible tag inside an article and check where the mouseup and mousedown end up 😄

Reply:
yes maybe i will assign id to every possible tag inside an article and check where the mouseup and mousedown end up 😄

Reply:
so i decided not to be too lazy ;D. Just add a new function regarding this issue